pandas/io/xml.py

```diff
@@ -408,9 +408,8 @@ def _iterparse_nodes(self) -> list[dict[str, str | None]]:
         for event, elem in iterparse(self.path_or_buffer, events=("start", "end")):
             curr_elem = elem.tag.split("}")[1] if "}" in elem.tag else elem.tag
 
-            if event == "start":
```

Comment:
I think this pattern was used on purpose. We are also using this in other places

Reply:
Oh okay, wasnt aware, i gave a quick look around some other code and i didnt catch it was a pattern